tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAwsVersioning.js

```diff
@@ -0,0 +1,548 @@
+            `deleting object, got error ${err}`);
+        _assertDeleteResult(result, resultType, versionId);
+        cb(null, result.VersionId);
+    });
+}
+
+function _createDeleteMarkers(s3, bucket, key, count, cb) {
+    return async.timesSeries(count,
+        (i, next) => delAndAssertResult(s3, { bucket, key,
+            resultType: newDeleteMarker }, next),
+        cb);
+}
+
+function _deleteDeleteMarkers(s3, bucket, key, deleteMarkerVids, cb) {
+    return async.mapSeries(deleteMarkerVids, (versionId, next) => {
+        delAndAssertResult(s3, { bucket, key, versionId,
+            resultType: deleteDeleteMarker }, next);
+    }, () => cb());
+}
+
+function _awsGetLatestVerId(key, body, cb) {
+    return awsS3.getObject({ Bucket: awsBucket, Key: key }, (err, result) => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `getting object from AWS, got error ${err}`);
+        const resultMD5 = expectedETag(result.Body, false);
+        const expectedMD5 = expectedETag(body, false);
+        assert.strictEqual(resultMD5, expectedMD5);
+        return cb(null, result.VersionId);
+    });
+}
+
+function _getAssertDeleted(s3, params, cb) {
+    const { key, versionId, errorCode } = params;
+    return s3.getObject({ Bucket: bucket, Key: key, VersionId: versionId },
+        err => {
+            assert.strictEqual(err.code, errorCode);
+            assert.strictEqual(err.statusCode, 404);
+            return cb();
+        });
+}
+
+function _awsGetAssertDeleted(params, cb) {
+    const { key, versionId, errorCode } = params;
+    return awsS3.getObject({ Bucket: awsBucket, Key: key, VersionId:
+        versionId }, err => {
+        assert.strictEqual(err.code, errorCode);
+        assert.strictEqual(err.statusCode, 404);
+        return cb();
+    });
+}
+
+describeSkipIfNotMultiple('AWS backend delete object w. versioning',
+function testSuite() {
+    this.timeout(30000);
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+        beforeEach(() => {
+            process.stdout.write('Creating bucket\n');
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return s3.createBucketAsync({ Bucket: bucket })
+            .catch(err => {
+                process.stdout.write(`Error creating bucket: ${err}\n`);
+                throw err;
+            });
+        });
+
+        afterEach(() => {
+            process.stdout.write('Emptying bucket\n');
+            return bucketUtil.empty(bucket)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(bucket);
+            })
+            .catch(err => {
+                process.stdout.write('Error emptying/deleting bucket: ' +
+                `${err}\n`);
+                throw err;
+            });
+        });
+
+        it('versioning not configured: should delete a specific version ' +
```

Comment:
versioning not configured on our s3: should put delete marker on key in AWS backend successfully